pkg/fleet-manager/backup_controller_helper.go

```diff
@@ -0,0 +1,490 @@
+		},
+	}
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+	}
+
+	// Iterate over each destination cluster
+	for _, cluster := range destinationClusters {
+		var listErr error
+		var deleteErr error
+
+		veleroClient := cluster.client.VeleroClient().VeleroV1()
+		switch resourceType {
+		case VeleroBackupKind:
+			var backupList *velerov1.BackupList
+			if backupList, listErr = veleroClient.Backups(VeleroNamespace).List(ctx, listOptions); listErr != nil {
+				return listErr
+			}
+			if len(backupList.Items) == 0 {
+				// All selected items are already deleted
+				return nil
+			}
+			// Delete all backups
+			for _, backup := range backupList.Items {
+				deleteErr = veleroClient.Backups(VeleroNamespace).Delete(ctx, backup.GetName(), metav1.DeleteOptions{})
+				if deleteErr != nil && !apierrors.IsNotFound(deleteErr) {
+					return deleteErr
+				}
+			}
+		case VeleroScheduleKind:
+			var scheduleList *velerov1.ScheduleList
+			if scheduleList, listErr = veleroClient.Schedules(VeleroNamespace).List(ctx, listOptions); listErr != nil {
+				return listErr
+			}
+			if len(scheduleList.Items) == 0 {
+				// All selected items are already deleted
+				return nil
+			}
+			// Delete all schedules
+			for _, schedule := range scheduleList.Items {
+				deleteErr = veleroClient.Schedules(VeleroNamespace).Delete(ctx, schedule.GetName(), metav1.DeleteOptions{})
+				if deleteErr != nil && !apierrors.IsNotFound(deleteErr) {
+					return deleteErr
+				}
+			}
+		case VeleroRestoreKind:
+			var restoreList *velerov1.RestoreList
+			if restoreList, listErr = veleroClient.Restores(VeleroNamespace).List(ctx, listOptions); listErr != nil {
+				return listErr
+			}
+			if len(restoreList.Items) == 0 {
+				// All selected items are already deleted
+				return nil
+			}
+			// Delete all restores
+			for _, restore := range restoreList.Items {
+				deleteErr = veleroClient.Restores(VeleroNamespace).Delete(ctx, restore.GetName(), metav1.DeleteOptions{})
+				if deleteErr != nil && !apierrors.IsNotFound(deleteErr) {
+					return deleteErr
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// if backup.Spec.Schedule is set, then it is ScheduleBackup. otherwise, it is regular/ont-time backup.
+func isScheduleBackup(backup *backupapi.Backup) bool {
+	return len(backup.Spec.Schedule) != 0
+}
+
+func generateVeleroInstanceLabel(createdByLabel, creatorName, fleetName string) map[string]string {
+	return map[string]string{
+		createdByLabel:  creatorName,
+		FleetLabel:      fleetName,
+		FleetPluginName: plugin.BackupPluginName,
+	}
+}
+
+func generateVeleroResourceObjectMeta(veleroResourceName string, labels map[string]string) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:      veleroResourceName,
+		Namespace: VeleroNamespace,
+		Labels:    labels,
+	}
+}
+
+func generateVeleroResourceName(clusterName, creatorKind, creatorName string) string {
+	return clusterName + "-" + creatorKind + "-" + creatorName
```

Comment:
clusterName is not a must

Reply:
Yes, the combination of creatorKind and creatorName can already identify a resource.

Reply:
because backup names stored in object storage must be unique, the clusterName is required.

here are examples of backup names in object storage if we are using the minimal backup example:

kurator-member1-backups-minimal
kurator-member2-backups-minimal

Reply:
IC